docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
```

Comment:
Agree strings are probably sufficient

On Fri, Jun 19, 2015 at 7:12 PM, Clayton Coleman <notifications@github.com>
wrote:

> In docs/design/configs.md
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6477#discussion_r32881104>
> :
>
> > @@ -0,0 +1,130 @@
> > +# Generic Config Object
> > +
> > +## Abstract
> > +
> > +A generic ``Config`` object to store command-line flags and environment variables
> > +that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
> > +is proposed in this proposal.
>
> I'm not aware of any form of configuration in common use that is primarily
> binary in nature, beyond encrypted and signed certificates which also have
> a common text form. Is ease of access worth potentially limiting some forms
> of config (gzipped config binary, protobuf serialization)? Most config
> seems like it would be classic text files or simple strings.
> … <#14e0ebcc861e3355_>
>  On Jun 19, 2015, at 7:22 PM, Tim Hockin <notifications@github.com>
> wrote: In docs/design/configs.md: > @@ -0,0 +1,130 @@ > +# Generic Config
> Object > + > +## Abstract > + > +A generic ``Config`` object to store
> command-line flags and environment variables > +that is used for
> configuration of ``Kubernetes`` components living on top of ``etcd`` > +is
> proposed in this proposal. isn't the current PR (map[string]string
> bascially exactly like secrets, except that secrets uses []byte as the
> value because binary. Do we want to support binary params? — Reply to this
> email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6477/files#r32881104>
> .
>